Helpers/PaymentPlansHelper.php

```diff
@@ -82,28 +93,117 @@ public function paymentTriggerIsAllowed():bool
+                $baseFeePlans[$planKey] = $feePlan;
+            }
+            $this->configHelper->saveBasePlansConfig($baseFeePlans);
+        } catch (RequestError $e) {
+            $this->logger->error('Error in save api base config plans', [$e->getMessage()]);
+        }
     }
 
     /**
-     * @return PaymentPlansConfigInterface
-     * @throws RequestError
+     * @param $plan
+     *
+     * @return array
      */
-    public function updatePlanConfigFromApi():PaymentPlansConfigInterface
+    private function forceAmountThresholds($plan): array
     {
-        $plansConfig = $this->createPlanConfig();
-        try {
-            $plansConfig->updateFromApi();
-        } catch (RequestError $e) {
-            $this->logger->info('Error fetching Alma payment plans : ',[$e->getMessage()]);
-            throw new RequestError("Error fetching Alma payment plans - displayed information might not be accurate");
+        $key = $plan['kind'].':'.$plan['installmentsCount'].':'.$plan['deferredDays'].':'.$plan['deferredMonths'];
+        if (
+            $plan[self::KEY_MIN_AMOUNT] < $plan[self::TRANSIENT_KEY_MIN_ALLOWED_AMOUNT] ||
+            $plan[self::KEY_MIN_AMOUNT] > $plan[self::TRANSIENT_KEY_MAX_ALLOWED_AMOUNT] ||
+            $plan[self::KEY_MIN_AMOUNT] > $plan[self::KEY_MAX_AMOUNT]
+        ) {
+            $plan[self::KEY_MIN_AMOUNT] = $plan[self::TRANSIENT_KEY_MIN_ALLOWED_AMOUNT];
+            $this->messageManager->addErrorMessage(
+                sprintf(__("Minimum amount is %s€ for plan %s"), ($plan[PaymentPlanConfig::TRANSIENT_KEY_MIN_ALLOWED_AMOUNT] / 100), $this->planLabelByKey($key))
+            );
         }
-        return $plansConfig;
+        if (
+            $plan[self::KEY_MAX_AMOUNT] > $plan[self::TRANSIENT_KEY_MAX_ALLOWED_AMOUNT] ||
+            $plan[self::KEY_MAX_AMOUNT] < $plan[self::TRANSIENT_KEY_MIN_ALLOWED_AMOUNT] ||
+            $plan[self::KEY_MAX_AMOUNT] < $plan[self::KEY_MIN_AMOUNT]
+        ) {
+            $plan[self::KEY_MAX_AMOUNT] = $plan[self::TRANSIENT_KEY_MAX_ALLOWED_AMOUNT];
+            $this->messageManager->addErrorMessage(
+                sprintf(__("Maximum amount is %s€ for plan %s"), ($plan[PaymentPlanConfig::TRANSIENT_KEY_MIN_ALLOWED_AMOUNT] / 100), $this->planLabelByKey($key))
+            );
+        }
+        return $plan;
     }
 
+    /**
+     * @param string $key
+     *
+     * @return string
+     */
+    public function planLabelByKey(string $key): string
+    {
+        preg_match('!general:([\d]+):([\d]+):([\d]+)!', $key, $matches);
+        $label = $key;
+
+        if (isset($matches[1])) {
+            $label =  __('Pay in %1 installments', $matches[1]);
+        }
+        if (isset($matches[2]) && $matches[2] != 0) {
+            $label =  __('Pay later - D+%1', $matches[2]);
+        }
+        return $label;
+    }
+
+    public function formatFeePlanConfigForBackOfficeDisplaying(FeePlan $feePlan, array $feePlanConfig = null): array
```

Comment:
```suggestion
    public function formatLocalFeePlanConfig(FeePlan $feePlan, array $feePlanConfig = null): array
```
It seems you want separate `feePlan` in to concepts:
1. `fromApi`
2. `forBackOffice` (that could become `fromBackOffice`)

Maybe you could call them:
1. `apiFeePlans`
2. and `localFeePlans` ?